app/models/publication_type.rb

```diff
@@ -1,143 +1,58 @@
-class PublicationType  < ActiveRecord::Base
-
+class PublicationType < ActiveRecord::Base
   has_many :publications
 
-  #this returns an instance of PublicationType according to one of the publication types
-  #if there is not a match nil is returned
-
-  #http://bib-it.sourceforge.net/help/fieldsAndEntryTypes.php#Entries
-
-  def self.for_type type
-    keys = { "Journal"=>"article",
-             "Book"=>"book",
-             "Booklet"=>"booklet",
-             "InBook"=>"inbook",
-             "InCollection"=>"incollection",
-             "InProceedings"=>"inproceedings",
-             "Manual"=>"manual",
-             "Masters_Thesis"=>"masters_thesis",
-             "Misc"=>"misc",
-             "Phd_Thesis"=>"phd_thesis",
-             "Bachelor_Thesis"=>"bachelorsthesis",
-             "Proceedings"=>"proceedings",
-             "Tech_Report"=>"tech_report",
-             "Unpublished"=>"unpublished" }
-    return PublicationType.find_by(key: keys[type])
-  end
-
-
 
+  # Map BibTeX keys → DataCite keys
+  BIBTEX_TO_DATACITE_KEY = {
+    'article' => 'journalarticle',
+    'book' => 'book',
+    'booklet' => 'booklet',
+    'inbook' => 'bookchapter',
+    'incollection' => 'collection',
+    'inproceedings' => 'conferencepaper',
+    'proceedings' => 'conferenceproceeding',
+    'manual' => 'text',
+    'misc' => 'other',
+    'unpublished' => 'preprint',
+    'techreport' => 'report',
+    'phdthesis' => 'phdthesis',
+    'mastersthesis' => 'mastersthesis',
+    'bachelorsthesis' => 'bachelorsthesis'
+  }.freeze
+
+
+  # Load expected publication types from YAML
+  def self.type_registry
+    @type_registry ||= begin
+                         path = Rails.root.join('config/default_data/publication_types.yml')
```

Comment:
don't use the yaml file, which is used for seeding. Fetch this info from the database where we know it is up to date